pkg/api/config.go

```diff
@@ -3,71 +3,177 @@ package api
+	} else {
+		validationErrors = append(validationErrors, errors.New("build_root is required"))
+	}
+
 	if config.Tests != nil {
-		for _, test := range config.Tests {
-			if test.As == "images" {
-				errs = append(errs, errors.New("test should not be called 'images' because it gets confused with '[images]' target"))
-			}
-			typeCount := 0
-			if testConfig := test.ContainerTestConfiguration; testConfig != nil {
-				typeCount += 1
-				// TODO remove when the migration is completed
-				if len(testConfig.From) == 0 && len(test.From) == 0 {
-					errs = append(errs, fmt.Errorf("test %q: `from` is required", test.As))
-				}
-			}
-			if testConfig := test.OpenshiftAnsibleClusterTestConfiguration; testConfig != nil {
-				typeCount += 1
-				errs = append(errs, validateTargetCloud(test.As, testConfig.TargetCloud))
-			}
-			if testConfig := test.OpenshiftAnsibleSrcClusterTestConfiguration; testConfig != nil {
-				typeCount += 1
-				errs = append(errs, validateTargetCloud(test.As, testConfig.TargetCloud))
-			}
-			if testConfig := test.OpenshiftInstallerClusterTestConfiguration; testConfig != nil {
-				typeCount += 1
-				errs = append(errs, validateTargetCloud(test.As, testConfig.TargetCloud))
+		validationErrors = append(validationErrors, validateTestStepConfiguration("tests", config.Tests))
+	}
+
+	if config.InputConfiguration.BaseImages != nil {
+		validationErrors = append(validationErrors, validateImageStreamTagReferenceMap("base_images", config.InputConfiguration.BaseImages))
+	}
+
+	if config.InputConfiguration.BaseRPMImages != nil {
+		validationErrors = append(validationErrors, validateImageStreamTagReferenceMap("base_rpm_images", config.InputConfiguration.BaseRPMImages))
+	}
+
+	if config.PromotionConfiguration != nil {
+		validationErrors = append(validationErrors, validatePromotionConfiguration("promotion", *config.PromotionConfiguration))
+	}
+
+	if config.InputConfiguration.ReleaseTagConfiguration != nil {
+		validationErrors = append(validationErrors, validateReleaseTagConfiguration("tag_specification", *config.InputConfiguration.ReleaseTagConfiguration))
+	}
+
+	return kerrors.NewAggregate(validationErrors)
+}
+
+func validateBuildRootImageConfiguration(fieldRoot string, input BuildRootImageConfiguration) error {
+	if input.ProjectImageBuild != nil && input.ImageStreamTagReference != nil {
+		return fmt.Errorf("%s: both image_stream_tag and project_image_build cannot be set", fieldRoot)
+	} else if input.ProjectImageBuild == nil && input.ImageStreamTagReference == nil {
+		return fmt.Errorf("%s: you have to specify either image_stream_tag or project_image_build", fieldRoot)
+	}
+	return nil
+}
+
+func validateTestStepConfiguration(fieldRoot string, input []TestStepConfiguration) error {
+	var validationErrors []error
+	typeCount := 0
+
+	// check for test.As duplicates
+	validationErrors = append(validationErrors, searchForTestDuplicates(input))
+
+	for num, test := range input {
+		if len(test.As) == 0 {
```

Comment:
We need to validate that this is a valid tag, so maybe just check for `[A-Za-z0-9-]`?